src/ast2ram/ClauseTranslator.cpp

```diff
@@ -139,14 +136,72 @@ Own<ram::Operation> ClauseTranslator::createProjection(const ast::Clause& clause
         values.push_back(ValueTranslator::translate(context, symbolTable, *valueIndex, arg));
     }
 
-    Own<ram::Operation> project = mk<ram::Project>(headRelationName, std::move(values));
+    VecOwn<ram::Condition> dependencies;
```

Comment:
Could you move this part out into its own method? e.g. `getFunctionalDependencies(...)`